prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelAddCommand.java

```diff
@@ -58,16 +69,42 @@ public ChannelAddCommand(CliConsole console, ActionFactory actionFactory) {
     public Integer call() throws Exception {
         Path installationDirectory = determineInstallationDirectory(directory);
         List<Repository> repositories;
-        repositories = RepositoryDefinition.from(repositoryDefs);
+        List<Channel> channels = null;
+            if(channelUrlOrPath.isPresent()) {
+                final ChannelCoordinate coord = ArtifactUtils.channelCoordFromString(channelUrlOrPath.get());
+                checkFileExists(coord.getUrl(), channelUrlOrPath.get());
+                ProvisioningDefinition provisioningDefinition = ProvisioningDefinition.builder()
+                        .setChannelCoordinates(channelUrlOrPath.get())
+                        .build();
+                channels = AbstractInstallCommand.resolveChannels(provisioningDefinition,
+                        MavenOptions.DEFAULT_OPTIONS);
+        } else if (!(gavUrlOrPath.isEmpty() || repositoryDefs.isEmpty())) {
+            repositories = RepositoryDefinition.from(repositoryDefs);
 
-        console.println(CliMessages.MESSAGES.subscribeChannel(installationDirectory, channelName));
+            console.println(CliMessages.MESSAGES.subscribeChannel(installationDirectory, channelName));
 
-        ChannelManifestCoordinate manifest = ArtifactUtils.manifestCoordFromString(gavUrlOrPath);
-        Channel channel = new Channel(channelName, null, null, repositories, manifest, null, null);
+            ChannelManifestCoordinate manifest = ArtifactUtils.manifestCoordFromString(gavUrlOrPath);
+            channels = List.of(new Channel(channelName, null, null, repositories, manifest, null, null));
+        } else {
+            return ReturnCodes.INVALID_ARGUMENTS;
```

Comment:
We should probably print some error message (`console.error(...)`) explaining that either manifest location or channel location must be set. We should also reject both of these parameters being set at once, they should be exclusive.